src/code/get_baseline.r

```diff
@@ -1,23 +1,37 @@
 library(epipredict)
 
-#' Return `date` if it has the desired weekday, else the next date that does
-#' @param date `Date` vector
-#' @param ltwday integerish vector; of weekday code(s), following POSIXlt
-#'   encoding but allowing either 0 or 7 to represent Sunday.
-#' @return `Date` object
-curr_or_next_date_with_ltwday <- function(date, ltwday) {
-  checkmate::assert_class(date, "Date")
-  checkmate::assert_integerish(ltwday, lower = 0L, upper = 7L)
-  date + (ltwday - as.POSIXlt(date)$wday) %% 7L
+parser <- argparser::arg_parser(
+  "Create a flat baseline model for covid-19 hospital admissions"
+)
+parser <- argparser::add_argument(
+  parser, "--reference-date",
+  help = "reference date in YYYY-MM-DD format"
+)
+
+args <- argparser::parse_args(parser)
+reference_date <- as.Date(args$reference_date)
+
+dow_supplied <- lubridate::wday(reference_date,
+  week_start = 7,
+  label = FALSE
+)
+if (dow_supplied != 7) {
+  cli::cli_abort(message = paste0(
+    "Expected `reference_date` to be a Saturday, day number 7 ",
+    "of the week, given the `week_start` value 7. ",
```

Comment:
```suggestion
    "of the week, given the `week_start` value of Sunday. ",
```
Since here it's fixed rather than dynamic, we can be clearer and say "Sunday"